### PR TITLE
fix: Redirect user to complete profile after first login

### PR DESCRIPTION
**Pull request by josebui**
## Description

- Redirect user to complete profile after first login
- Stored a new value in local storage to know if the user has already viewed the profile page after the first login

### Checklist
- [x] Corresponding issue has been opened
- [x] New tests added

### Dependency
- https://github.com/techmatters/terraso-backend/pull/1195

### Related Issues
- https://github.com/techmatters/terraso-web-client/issues/1145

### CODE REVIEW

File: src/account/components/AccountProfile.js [hunk 1]
```diff
@@ -170,13 +178,21 @@ const AccountProfile = () => {
...
   useDocumentTitle(t('account.profile_document_title'));
   useDocumentDescription(t('account.profile_document_description'));
 
+  useEffect(
+    () => () => {
+      profileCompleted(user?.email);
+    },
+    [user?.email]
+  );
+
```
Comment (paulschreiber) [inline src/account/components/AccountProfile.js:195]: question: why is this doubly nested?

**Review by paulschreiber — APPROVED** [review]

**Review by paulschreiber — CHANGES_REQUESTED** [review]
@josebui Please add the missing translation.

File: src/localization/locales/es-ES.json [hunk 2]
```diff
@@ -219,6 +219,7 @@
         "profile_document_title": "Cuenta",
         "profile_document_description": "Administrar las preferencias de su perfil de Terraso",
         "profile_form_label": "Administrar perfil",
+        "profile_complete_message": "Sólo necesitamos algunos detalles sobre ti antes de que poder comenzar.",
```
Comment (paulschreiber) [inline src/localization/locales/es-ES.json:222]:
```suggestion
        "profile_complete_message": "Necesitamos algunos detalles sobre ti antes de que poder comenzar.",
```

**Review by paulschreiber — APPROVED** [review]

File: src/localization/locales/en-US.json [hunk 3]
```diff
@@ -219,6 +219,7 @@
         "profile_document_title": "Account",
         "profile_document_description": "View and edit your Terraso profile and preferences",
         "profile_form_label": "Manage profile",
+        "profile_complete_message": "We just need a few details about you before you can get started.",
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:222]: Let us know what to call you, and what language you’d like to use.

Comment (paulschreiber) [inline src/localization/locales/en-US.json:222]:
```suggestion
        "profile_complete_message": "Tell us what to call you and review your profile settings.",
```

**Review by paulschreiber — APPROVED** [review]

**Review by paulschreiber — APPROVED** [review]

File: src/account/components/AccountProfile.js [hunk 4]
```diff
@@ -191,27 +210,40 @@ const AccountProfile = () => {
...
+        return dispatch(
+          savePreference({ key: preferenceKey, value: newValue.toString() })
+        );
+      }
+      return null;
+    });
+
+    const allPromises = [saveUserPromise, ...savePreferencesPromises].filter(
+      promise => !!promise
```
Comment (paulschreiber) [inline src/account/components/AccountProfile.js:239]:
```suggestion
      promise => Boolean(promise)
```

File: src/account/accountProfileUtils.js [hunk 5]
```diff
@@ -73,7 +73,7 @@ export const useCompleteProfile = () => {
       return;
     }
 
-    if (location.pathname === '/account/profile/completeProfile') {
+    if (location?.pathname.includes('/account/profile/completeProfile')) {
```
[the remark this answers is not in the document]
Comment (josebui) [reply src/account/accountProfileUtils.js:76]: I think includes makes more sense

**Review by paulschreiber — APPROVED** [review]

**Pull request merged**